core/dbt/parser/models.py

```diff
@@ -233,6 +234,26 @@ def render_update(self, node: ParsedModelNode, config: ContextConfig) -> None:
                 raise ParsingException(msg, node=node) from exc
             return
 
+        if node.language == ModelLanguage.prql:
+            provider = PrqlProvider()
+            try:
+                context = self._context_for(node, config)
+                references = provider.list_references(node.raw_code)
+                sql = provider.compile(node.raw_code, references)
```

Comment:
After taking another look here — with plans to do some hacking on/around this soon! — I think we'd want the actual `.compile()` call (PRQL &rarr; SQL transpilation) to happen within compilation, rather than during parsing. [Here's where we do that for Python.](https://github.com/dbt-labs/dbt-core/blob/517576c088c6fa54c5fa510c455b924d735d63f5/core/dbt/compilation.py#L372-L390)

I think the goal here, within the parser, should just be to provide the set of `references`, and then pick up again at compile time. Something to also think about: Certain modeling languages will want a database connection at compile time (e.g. Jinja-SQL, Cody's prototype work on Ibis); some won't need one (e.g. Snowpark/PySpark Python, PRQL), and so shouldn't require one.

Reply:
OK great, that seems very reasonable.

Would you have it as an `if` in `Compiler` like the current python implementation, or inherit a class for each lang; i.e. `CompilerPython` / `CompilerPrql`?